Build fails without default features

**Describe the bug**
Build fails on main. Git blame points to https://github.com/tracel-ai/burn/commit/caba944bd049cd4e1b3d2162c6db6db6c3986a4c. Enabling "autotune" in `burn` makes the compile succeed.

**To Reproduce**
- `cargo init`

- Add the following dependencies:
```
burn-train = { git = "https://github.com/tracel-ai/burn.git", branch = "main", default-features = false }

[dependencies.burn]
git = "https://github.com/tracel-ai/burn.git"
branch = "main"
default-features = false
features = ["std", "fusion", "ndarray", "webgpu", "cuda", "autodiff"]
```

- `cargo build`


**Desktop (please complete the following information):**
 - OS: Linux
 - Rust version: 1.91.1

**Additional context**
```
error[E0412]: cannot find type `Simple` in this scope
   --> /home/user/.cargo/git/checkouts/burn-6c277d792b0d5d7a/ddb8de6/crates/burn-cubecl-fusion/src/optim/matmul/optimization.rs:172:36
    |
172 |         if arg.execute_fused::<BT, Simple>(context).is_err() {
    |                                    ^^^^^^ not found in this scope
    |
help: there is an enum variant `crate::optim::matmul::FusedMatmulSelector::Simple`; try using the variant's enum
    |
172 -         if arg.execute_fused::<BT, Simple>(context).is_err() {
172 +         if arg.execute_fused::<BT, crate::optim::matmul::FusedMatmulSelector>(context).is_err() {
    |
help: you might be missing a type parameter
    |
133 | impl<R: Runtime, Simple> MatmulOptimization<R> {
    |                ++++++++

error[E0107]: method takes 1 generic argument but 2 generic arguments were supplied
   --> /home/user/.cargo/git/checkouts/burn-6c277d792b0d5d7a/ddb8de6/crates/burn-cubecl-fusion/src/optim/matmul/optimization.rs:172:16
    |
172 |         if arg.execute_fused::<BT, Simple>(context).is_err() {
    |                ^^^^^^^^^^^^^     -------- help: remove the unnecessary generic argument
    |                |
    |                expected 1 generic argument
    |
note: method defined here, with 1 generic parameter: `BT`
   --> /home/user/.cargo/git/checkouts/burn-6c277d792b0d5d7a/ddb8de6/crates/burn-cubecl-fusion/src/optim/matmul/optimization.rs:87:19
    |
 87 |     pub(crate) fn execute_fused<BT: CubeElement>(
    |                   ^^^^^^^^^^^^^ --

error[E0061]: this method takes 2 arguments but 1 argument was supplied
   --> /home/user/.cargo/git/checkouts/burn-6c277d792b0d5d7a/ddb8de6/crates/burn-cubecl-fusion/src/optim/matmul/optimization.rs:172:16
    |
172 |         if arg.execute_fused::<BT, Simple>(context).is_err() {
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^--------- argument #2 of type `optim::matmul::optimization::FusedMatmulSelector` is missing
    |
note: method defined here
   --> /home/user/.cargo/git/checkouts/burn-6c277d792b0d5d7a/ddb8de6/crates/burn-cubecl-fusion/src/optim/matmul/optimization.rs:87:19
    |
 87 |     pub(crate) fn execute_fused<BT: CubeElement>(
    |                   ^^^^^^^^^^^^^
...
 90 |         selector: FusedMatmulSelector,
    |         -----------------------------
help: provide the argument
    |
172 |         if arg.execute_fused::<BT, Simple>(context, /* optim::matmul::optimization::FusedMatmulSelector */).is_err() {
    |                                                   ++++++++++++++++++++++++++++++++++++++++++++++++++++++++

error[E0599]: no method named `execute_fused_reduce` found for struct `ReduceOptimizationTuneArg<R>` in the current scope
   --> /home/user/.cargo/git/checkouts/burn-6c277d792b0d5d7a/ddb8de6/crates/burn-cubecl-fusion/src/optim/reduce/optimization.rs:211:16
    |
 49 | pub(crate) struct ReduceOptimizationTuneArg<R: Runtime> {
    | ------------------------------------------------------- method `execute_fused_reduce` not found for this struct
...
211 |         if arg.execute_fused_reduce::<BT>(context).is_err() {
    |                ^^^^^^^^^^^^^^^^^^^^
    |
help: there is a method `execute_fused` with a similar name, but with different arguments
   --> /home/user/.cargo/git/checkouts/burn-6c277d792b0d5d7a/ddb8de6/crates/burn-cubecl-fusion/src/optim/reduce/optimization.rs:121:5
    |
121 | /     pub fn execute_fused<BT: CubeElement>(
122 | |         &self,
123 | |         context: &mut Context<'_, CubeFusionHandle<R>>,
124 | |         strategy: ReduceStrategy,
125 | |     ) -> Result<TuneOutput<R>, TraceError<FusedReduceError>> {
    | |____________________________________________________________^

Some errors have detailed explanations: E0061, E0107, E0412, E0599.
For more information about an error, try `rustc --explain E0061`.
error: could not compile `burn-cubecl-fusion` (lib) due to 4 previous errors
```

--- a/crates/burn-cubecl-fusion/src/optim/matmul/optimization.rs
+++ b/crates/burn-cubecl-fusion/src/optim/matmul/optimization.rs
@@ -1,4 +1,5 @@
 use super::args::FusedMatmulInputLaunch;
+#[cfg(feature = "autotune")]
 use super::tune::fused_matmul_autotune;
 use crate::{
     CubeFusionHandle, FallbackOperation,
@@ -169,7 +170,10 @@ impl<R: Runtime> MatmulOptimization<R> {
         fused_matmul_autotune::<R, BT>(arg, context);
 
         #[cfg(not(feature = "autotune"))]
-        if arg.execute_fused::<BT, Simple>(context).is_err() {
+        if arg
+            .execute_fused::<BT>(context, FusedMatmulSelector::default())
+            .is_err()
+        {
             arg.execute_fallback::<BT>(context);
         }
     }

--- a/crates/burn-cubecl-fusion/src/optim/reduce/optimization.rs
+++ b/crates/burn-cubecl-fusion/src/optim/reduce/optimization.rs
@@ -1,6 +1,7 @@
 use super::args::{
     FusedReduceInput, FusedReduceInputLaunch, FusedReduceOutput, FusedReduceOutputLaunch,
 };
+#[cfg(feature = "autotune")]
 use super::tune::fused_reduce_autotune;
 use crate::engine::launch::FuseTraceLauncher;
 use crate::engine::launch::runner::{TraceRunner, Vectorization};
@@ -208,7 +209,16 @@ impl<R: Runtime> ReduceOptimization<R> {
         fused_reduce_autotune::<R, BT>(arg, context);
 
         #[cfg(not(feature = "autotune"))]
-        if arg.execute_fused_reduce::<BT>(context).is_err() {
+        if arg
+            .execute_fused::<BT>(
+                context,
+                ReduceStrategy {
+                    shared: false,
+                    use_planes: false,
+                },
+            )
+            .is_err()
+        {
             arg.execute_fallback::<BT>(context);
         }
     }

